src/serialization/Portable.ts

```diff
@@ -41,127 +48,552 @@ export enum FieldType {
 
-    writeFloatArray(fieldName: string, floats: number[]): void;
+    /**
+     * Writes a primitive boolean-array.
+     *
+     * @param fieldName name of the field
+     * @param booleans  boolean array to be written
+     */
+    writeBooleanArray(fieldName: string, booleans: boolean[] | null): void;
 
-    writeShortArray(fieldName: string, shorts: number[]): void;
+    /**
+     * Writes a primitive char-array.
+     *
+     * @param fieldName name of the field
+     * @param chars     char array to be written
+     */
+    writeCharArray(fieldName: string, chars: string[] | null): void;
+
+    /**
+     * Writes a primitive int-array.
+     *
+     * @param fieldName name of the field
+     * @param ints      int array to be written
+     */
+    writeIntArray(fieldName: string, ints: number[] | null): void;
+
+    /**
+     * Writes a primitive long-array.
+     *
+     * @param fieldName name of the field
+     * @param longs     long array to be written
+     */
+    writeLongArray(fieldName: string, longs: Long[] | null): void;
+
+    /**
+     * Writes a primitive double array.
+     *
+     * @param fieldName name of the field
+     * @param values    double array to be written
+     */
+    writeDoubleArray(fieldName: string, values: number[] | null): void;
+
+    /**
+     * Writes a primitive float array.
+     *
+     * @param fieldName name of the field
+     * @param values    float array to be written
+     */
+    writeFloatArray(fieldName: string, values: number[] | null): void;
+
+    /**
+     * Writes a primitive short-array.
+     *
+     * @param fieldName name of the field
+     * @param values    short array to be written
+     */
+    writeShortArray(fieldName: string, values: number[] | null): void;
+
+    /**
+     * Writes a String-array.
+     *
+     * @param fieldName name of the field
+     * @param values    String array to be written
+     * @deprecated  since version 4.2. for the sake of better naming. Please use {@link writeStringArray} instead.
+     * This method will be removed in next major version.
+     */
+    writeUTFArray(fieldName: string, values: string[] | null): void;
+
+    /**
+     * Writes a String-array.
+     *
+     * @param fieldName name of the field
+     * @param values    String array to be written
+     */
+    writeStringArray(fieldName: string, values: string[] | null): void;
+
+    /**
+     * Writes a an array of Portables.
+     *
+     * @param fieldName name of the field
+     * @param values    portable array to be written
+     */
+    writePortableArray(fieldName: string, values: Portable[] | null): void;
+
+    /**
+     * Writes an array of Decimals
+     *
+     * @param fieldName name of the field
+     * @param values    BigDecimal array to be written
+     * @see {@link writeDecimal}
+     */
+    writeDecimalArray(fieldName: string, values: BigDecimal[] | null): void;
+
+    /**
+     * Writes an array of Time's
```

Comment:
```suggestion
     * Writes an array of LocalTime's
```

Same for others. I think we should use the full class name of the thing we write